github_poster/loader/spotify_loader.py

```diff
@@ -0,0 +1,153 @@
+import datetime
+import json
+import os
+import time
+from base64 import urlsafe_b64encode
+from collections import defaultdict
+
+import requests
+
+from github_poster.loader.base_loader import BaseLoader, LoadError
+from github_poster.loader.config import SPOTIFY_HISTORY_URL, SPOTIFY_TOKEN
+
+
+class SpotifyLoader(BaseLoader):
+    track_color = "#FFE411"
+    unit = "songs"
+
+    def __init__(self, from_year, to_year, _type, **kwargs):
+        super().__init__(from_year, to_year, _type)
+        self.number_by_date_dict = defaultdict(int)
+        self.session = requests.Session()
+        self.client_id = kwargs.get("spotify_client_id", "")
+        self.client_secret = kwargs.get("spotify_client_secret", "")
+        self.refresh_token = kwargs.get("spotify_refresh_token", "")
+        self.access_token = ""
+        self.spotify_file = kwargs.get("spotify_history_file")
+        self._parse_spotify_history()
+
+    @classmethod
+    def add_loader_arguments(cls, parser, optional):
+        parser.add_argument(
+            "--spotify_client_id",
+            dest="spotify_client_id",
+            type=str,
+            required=optional,
+            help="",
+        )
+        parser.add_argument(
+            "--spotify_client_secret",
+            dest="spotify_client_secret",
+            type=str,
+            required=optional,
+            help="",
+        )
+        parser.add_argument(
+            "--spotify_refresh_token",
+            dest="spotify_refresh_token",
+            type=str,
+            required=optional,
+            help="",
+        )
+        parser.add_argument(
+            "--spotify_history_file",
+            dest="spotify_history_file",
+            type=str,
+            default=os.path.join("IN_FOLDER", "spotify-history.json"),
+            help="spotify history file path",
+        )
+
+    def _parse_spotify_history(self):
+        if os.path.exists(self.spotify_file):
+            with open(self.spotify_file, "r") as f:
+                self.number_by_date_dict = json.load(f)
+
+    def _writeback_spotify_history(self):
+        with open(self.spotify_file, "w") as f:
+            json.dump(self.number_by_date_dict, f, sort_keys=True)
```

Comment:
This two functions seems same as `bilibili` can we refactor them to one?

Reply:
Maybe I can put it in `BaseLoader`. When needed in SpotifyLoader, use `self._type` to get the name of history file. If other loader needs the function to save history data, call it in the same way.

Reply:
its fine.